scripts/performance_test_annotations_backend.py

```diff
@@ -0,0 +1,200 @@
+            dict_size = sum(sys.getsizeof(value) for value in annon_dict.values()) / 1024 ** 2
+            self.test_notes[dataset_name][f"num_categories_{num_cat}"][f"max_label_{max_labels}"]["annotation_dict"] = {
+                "creation_time": str(elapsed()),
+                "size": f"{dict_size} mb",
+            }
+            df = pd.DataFrame(annon_dict)
+            df_size = sys.getsizeof(df) / 1024 ** 2
+            self.test_notes[dataset_name][f"num_categories_{num_cat}"][f"max_label_{max_labels}"]["data_frame"] = {
+                "creation_time": str(elapsed()),
+                "size": f"{df_size} mb",
+            }
+            try:
+                matrix = encode_matrix_fbs(matrix=df, row_idx=None, col_idx=df.columns)
+                matrix_size = sys.getsizeof(matrix) / 1024 ** 2
+                self.test_notes[dataset_name][f"num_categories_{num_cat}"][f"max_label_{max_labels}"]["fbs_matrix"] = {
+                    "creation_time": str(elapsed()),
+                    "size": f"{matrix_size} mb",
+                }
+                return matrix
+            except Exception as e:
+                print(f"Issue creating fbs matrix: {e}, for {dataset_name}")
+                return []
+
+    def send_put_request(self, dataset_url, data):
+        if not data:
+            return [], 0.0
+        url = self.url_base + f"{dataset_url}/api/v0.2/annotations/obs"
+        with self.elapsed_timer() as elapsed:
+            try:
+                headers = {"Content-Type": "application/octet-stream", "Cookie": cookie}
+                response = requests.put(url=url, data=data, headers=headers)
+            except Exception as e:
+                print(f"Issue with put request: {e}")
+                return None, elapsed()
+            return response, elapsed()
+
+    def test_categories_max_label_matrix(self, dataset_name):
+        for unique_labels in self.max_labels:
+            for category_count in self.annotations_category_count:
+                print(f"Starting dataset: {dataset_name}, categories: {category_count}, labels: {unique_labels}")
+                fbs_matrix = self.create_matrix(dataset_name, category_count, unique_labels)
+                response, response_time = self.send_put_request(
+                    self.test_datasets[dataset_name]["dataset_url"], fbs_matrix
+                )
+                if response is None:
+                    self.test_notes[dataset_name][f"num_categories_{category_count}"][f"max_label_{unique_labels}"][
+                        "put_request"
+                    ] = {"response_status": "failed", "request_time": str(response_time)}
+                else:
+                    self.test_notes[dataset_name][f"num_categories_{category_count}"][f"max_label_{unique_labels}"][
+                        "put_request"
+                    ] = {"response_status": response.status_code, "request_time": str(response_time)}
+
+
+def test_all_datasets():
```

Comment:
What calls test_all_datasets?
I think there is something missing to run the tests.

Reply:
For now I think the tests will only be run manually. You still need to generate a cookie and update that in the code. I tried using the machine to machine tokens from auth0 but they run into issue bc they dont have a user profile. 

I can add a command to the Makefile, I just wanted to be sure people knew they needed to create a cookie